Update from legacy circleci images to cimg.

leapingfrogs opened the pull request:
The circle ci image being used is being [deprecated](https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034).

This simply upgrades to using the latest ruby orb and now runs against ruby versions '2.6', '2.7', '3.0', '3.1' using the newer [convenience images](https://circleci.com/blog/announcing-our-next-generation-convenience-images-smaller-faster-more-deterministic/) from circleci.

leapingfrogs committed:
Update from legacy circleci images to cimg.

emancu commented on .circleci/config.yml:
```diff
@@ -18,4 +21,7 @@ workflows:
+      - build:
+          matrix:
+            parameters:
+              ruby: ['2.7', '3.0']
```
```suggestion
              ruby: ['2.6', '2.7', '3.0', '3.1']
```

leapingfrogs committed:
Update .circleci/config.yml

Co-authored-by: Emiliano Mancuso <emiliano.mancuso@carwow.co.uk>

leapingfrogs committed:
New images already have bundler installed

leapingfrogs committed:
Update apt-cache to access unixodbc-dev

As per guidance in: https://circleci.com/docs/2.0/next-gen-migration-guide/#troubleshooting
